### PR TITLE
 RSS-ECOMM-3_09-10 Implement Detailed Product Information Retrieval and Image Slider

### PR DESCRIPTION
**Pull request by aleks6699**
Was done:
- The application successfully retrieves detailed product information from the commercetools API, including basic information: 
   name, description, images and other attributes
- The resulting product information is displayed on the Product Details page in an organized and clear way
- The application successfully retrieves multiple product images from the selected API if available.
   The image slider is implemented on the product detail page and can handle multiple images. A slider allows users to manually control which image is displayed.

### CODE REVIEW

**Review by MartiP54 — APPROVED** [review]
Great job !

**Review by AnastasiiaRamona — APPROVED** [review]
In the following branches, it is recommended to change the styles in accordance with the overall window size. Good job!

File: src/components/detailedProduct/detailedProductView.ts [hunk 1]
```diff
@@ -1,52 +1,37 @@
...
-  price: string = 'Price';
+  id: string;
 
-  discountPrice: string = 'Discount Price';
-
-  descriptionOfProduct: string = 'Description';
+  constructor(id: string) {
+    this.controller = new DetailedProductController();
+    this.id = id;
+  }
```
Comment (AnastasiiaRamona) [inline src/components/detailedProduct/detailedProductView.ts:12]: Initializing default values for the properties ensures that the object is in a valid state upon instantiation. The constructor properly initializes the controller and ID.

File: src/components/detailedProduct/detailedProductView.ts [hunk 2]
```diff
@@ -1,52 +1,37 @@
...
 
   renderMain() {
     const detailedProductWrapper = HTMLCreator.createElement('div', { class: 'detailed__product-wrapper' });
     const main = HTMLCreator.createElement('main', { class: 'detailed__product' }, [
       HTMLCreator.createElement('div', { class: 'detailed__product-container' }, [detailedProductWrapper]),
     ]);
```
Comment (AnastasiiaRamona) [inline src/components/detailedProduct/detailedProductView.ts:18]: The method renderMain is constructing HTML elements effectively. Consider breaking down the creation of complex elements into smaller helper methods to improve readability.

File: src/components/detailedProduct/detailedProductView.ts [hunk 3]
```diff
@@ -1,52 +1,37 @@
...
-      HTMLCreator.createElement('p', { class: 'detailed__product-description' }, [this.descriptionOfProduct]),
-      HTMLCreator.createElement('div', { class: 'detailed__product-price' }, [this.price]),
-      HTMLCreator.createElement('div', { class: 'detailed__product-discount' }, [this.discountPrice]),
+      HTMLCreator.createElement('h1', { class: 'detailed__product-title' }),
+      HTMLCreator.createElement('p', { class: 'detailed__product-description' }),
+      HTMLCreator.createElement('div', { class: 'detailed__product-inner' }, [
+        HTMLCreator.createElement('div', { class: 'detailed__product-price' }),
+        HTMLCreator.createElement('div', { class: 'detailed__product-discount' }),
+      ]),
+
```
Comment (AnastasiiaRamona) [inline src/components/detailedProduct/detailedProductView.ts:34]: The productContent is built clearly and concisely. There are redundant elements being created but not utilized. Ensure all created elements are necessary for the rendering logic.

File: src/components/detailedProduct/detailedProductView.ts [hunk 4]
```diff
@@ -55,4 +40,29 @@ export default class DetailedProduct {
...
+    const result = await this.controller.getProductByID(this.id);
+    if (result) {
+      const productData = {
+        name: result.body.masterData.current.name['en-US'],
+        description: result.body.masterData.current.description?.['en-US'],
+        images: result.body.masterData.current.masterVariant.images?.map((image) => image.url),
+        price: result.body.masterData.current.masterVariant.prices?.[0]?.value.centAmount,
+        discountedPrice: result.body.masterData.current.masterVariant.prices?.[0]?.discounted?.value.centAmount,
+      };
+      console.log(result);
```
Comment (AnastasiiaRamona) [inline src/components/detailedProduct/detailedProductView.ts:54]: The getProductInformation method is efficiently fetching and processing product data. Consider removing the console.log(result) statement before production deployment.

**Pull request merged**